node/src/components/linear_chain.rs

```diff
@@ -129,16 +125,13 @@ where
             Event::Request(LinearChainRequest::BlockRequest(block_hash, sender)) => effect_builder
                 .get_block_from_storage(block_hash)
                 .event(move |maybe_block| Event::GetBlockResult(block_hash, maybe_block, sender)),
-            Event::Request(LinearChainRequest::LastFinalizedBlock(responder)) => {
-                responder.respond(self.last_block.clone()).ignore()
-            }
```

Comment:
I would prefer if the requests didn't go to the storage directly. I think it's bad design. `LinearChain` component is an aggregate root for the linear chain information. It knows what's valid and what is invalid. It decides when to store block as a linear chain block or not. It should also answer queries b/c no other component should even know what the linear chain block is.

Reply:
Well, we can rejig the flow if required, but I don't really see much point to calls going from ApiServer via LinearChain to Storage personally.  There's no real risk as far as I can see, and it just seems like building in latency and complexity for no gain.

Reply:
Actually, it is better in terms of performance b/c it uses the cached LFB instead of reading it from the storage.

Reply:
Yes, that could be true, but I believe caching the LFB in the linear chain was always deemed a temporary workaround.  One negative of using the cached block from the linear chain is that it never gets updated with the proofs.  With this change, the user gets back the actual block as stored - i.e. it will include all known proofs.

Reply:
> but I believe caching the LFB in the linear chain was always deemed a temporary workaround. 

Well, not for me at least. I think it's perfectly fine to cache LFB in the linear chain component as it's the place where new linear chain blocks should be inserted into the storage anyway.

> One negative of using the cached block from the linear chain is that it never gets updated with the proofs. With this change, the user gets back the actual block as stored - i.e. it will include all known proofs.

Then finality signatures for cached LFB should also be updated in the Linear Chain component. It already handles `Event::NewFinalitySignature` event and updates that stored LFB itself.